### PR TITLE
Updated version of PR 841

### PR DESCRIPTION
**Pull request by demobox**
See #841. I ran the following tests on this variant:

* `mvn duplicate-finder:check` from the jclouds root is successful
* `mvn dependency:tree` in drivers/sshj does not show any `net.schmizz` packages, and has the correct bouncycastle deps

```
[INFO] ------------------------------------------------------------------------
[INFO] Building jclouds sshj ssh client 2.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ jclouds-sshj ---
[INFO] org.apache.jclouds.driver:jclouds-sshj:bundle:2.0.0-SNAPSHOT
[INFO] +- org.apache.jclouds:jclouds-compute:jar:2.0.0-SNAPSHOT:compile
[INFO] |  \- org.apache.jclouds:jclouds-scriptbuilder:jar:2.0.0-SNAPSHOT:compile
[INFO] +- org.apache.jclouds:jclouds-core:jar:2.0.0-SNAPSHOT:compile
[INFO] |  +- javax.ws.rs:jsr311-api:jar:1.1.1:compile
[INFO] |  +- com.google.inject.extensions:guice-assistedinject:jar:3.0:compile
[INFO] |  +- com.google.inject:guice:jar:3.0:compile
[INFO] |  |  \- aopalliance:aopalliance:jar:1.0:compile
[INFO] |  +- javax.inject:javax.inject:jar:1:compile
[INFO] |  +- javax.annotation:jsr250-api:jar:1.0:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.3.1:compile
[INFO] |  \- com.google.guava:guava:jar:16.0.1:compile
[INFO] +- org.apache.jclouds.driver:jclouds-slf4j:jar:2.0.0-SNAPSHOT:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.2:compile
[INFO] +- ch.qos.logback:logback-classic:jar:1.1.2:test
[INFO] |  \- ch.qos.logback:logback-core:jar:1.1.2:test
[INFO] +- org.apache.jclouds.driver:jclouds-bouncycastle:jar:2.0.0-SNAPSHOT:compile
[INFO] |  \- org.bouncycastle:bcprov-ext-jdk15on:jar:1.51:compile <<<<<<<
[INFO] +- org.apache.jclouds:jclouds-core:test-jar:tests:2.0.0-SNAPSHOT:test
[INFO] +- com.hierynomus:sshj:jar:0.12.0:compile <<<<<<<
[INFO] |  +- org.bouncycastle:bcpkix-jdk15on:jar:1.51:compile <<<<<<<
[INFO] |  \- com.jcraft:jzlib:jar:1.1.3:compile
[INFO] +- com.jcraft:jsch.agentproxy.sshj:jar:0.0.9:compile <<<<<<<
[INFO] |  \- com.jcraft:jsch.agentproxy.core:jar:0.0.9:compile
[INFO] +- com.jcraft:jsch.agentproxy.connector-factory:jar:0.0.9:compile <<<<<<<
[INFO] |  +- com.jcraft:jsch.agentproxy.usocket-jna:jar:0.0.9:compile
[INFO] |  |  +- net.java.dev.jna:jna:jar:4.1.0:compile
[INFO] |  |  \- net.java.dev.jna:jna-platform:jar:4.1.0:compile
[INFO] |  +- com.jcraft:jsch.agentproxy.usocket-nc:jar:0.0.9:compile
[INFO] |  +- com.jcraft:jsch.agentproxy.sshagent:jar:0.0.9:compile
[INFO] |  \- com.jcraft:jsch.agentproxy.pageant:jar:0.0.9:compile
[INFO] +- org.testng:testng:jar:6.8.8:test
[INFO] |  +- org.beanshell:bsh:jar:2.0b4:test
[INFO] |  \- com.beust:jcommander:jar:1.27:test
[INFO] +- org.easymock:easymockclassextension:jar:3.1:test
[INFO] |  \- org.easymock:easymock:jar:3.1:test
[INFO] |     +- cglib:cglib-nodep:jar:2.2.2:test
[INFO] |     \- org.objenesis:objenesis:jar:1.2:test
[INFO] +- xmlunit:xmlunit:jar:1.3:test
[INFO] +- org.clojure:clojure:jar:1.3.0:compile
[INFO] +- org.clojure:tools.logging:jar:0.2.3:compile
[INFO] +- org.clojure:core.incubator:jar:0.1.0:compile
[INFO] +- org.assertj:assertj-core:jar:1.7.0:test
[INFO] \- org.assertj:assertj-guava:jar:1.3.0:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

@steveloughran @nacx Ping ;-)

### CODE REVIEW

File: drivers/sshj/pom.xml [hunk 1]
```diff
@@ -87,15 +87,9 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>net.schmizz</groupId>
+      <groupId>com.hierynomus</groupId>
       <artifactId>sshj</artifactId>
-      <version>0.8.1</version>
-    </dependency>
-    <!-- required by sshj -->
-    <dependency>
```
Comment (demobox) [inline drivers/sshj/pom.xml]: No longer needed as a separate dep, as sshj now [includes it](https://github.com/hierynomus/sshj/blob/v0.12.0/pom.xml#L96-L99)

File: drivers/sshj/pom.xml [hunk 2]
```diff
@@ -107,12 +101,19 @@
...
+        <!-- required due to sshj group ID change -->
+        <exclusion>
+          <groupId>net.schmizz</groupId>
+          <artifactId>sshj</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>com.jcraft</groupId>
       <artifactId>jsch.agentproxy.connector-factory</artifactId>
```
Comment (demobox) [inline drivers/sshj/pom.xml]: Works for me (famous last words ;-)) without an exclusion here

File: project/pom.xml [hunk 3]
```diff
@@ -572,23 +572,6 @@
                 <package>com.google</package>
               </packages>
             </exception>
-            <exception>
```
Comment (demobox) [inline project/pom.xml]: Doesn't seem to be necessary any more? Ran fine locally, at least - let's see what the PR builder has to say...

**Comment by demobox** [comment]
> 1 failing check

Looks like the duplicate finder exception is needed after all - some local environment thing. Glad we have the PR builder setup!

**Comment by demobox** [comment]
> 1 successful check

**That's** better ;-)

**Comment by nacx** [comment]
I assume the dependency:tree of the PR description is outdated now? It shows the non-extended BC dependency, but I see it excluded in the pom.xml

**Comment by demobox** [comment]
> It shows the non-extended BC dependency, but I see it excluded in the pom.xml

Thanks for checking, @nacx! The dependency:tree output should be up to date. `com.hierynomus:sshj` pulls in `bcpkix-jdk15on:jar:1.51`, and `jclouds-bouncycastle` pulls in `bcprov-ext-jdk15on:jar`. Are you expecting/do we need something different here?

Full output of a "fresh" `mvn dependency:tree` run [here](http://pastie.org/private/7yo7xeavjniypuzlsiqyya).

**Comment by nacx** [comment]
Oh,I see now. It pulls **bcpkix**-jdk15on:jar:1.51 instead of **bcprov**-jdk15on:jar:1.51, so I assume there aren't conflicting classes witht he extended jar. lgtm!

**Comment by demobox** [comment]
> so I assume there aren't conflicting classes witht he extended jar

I don't think so - these are two separate deps, indeed. We were pulling both in previously without the need for a duplicate-finder exception, so I think we should be OK. Thanks for checking!

**Comment by demobox** [comment]
> lgtm!

Thanks, Ignasi! Just squashed the commits. One more PR builder test, just to be sure.

**Comment by demobox** [comment]
Committed to [master](https://git-wip-us.apache.org/repos/asf?p=jclouds.git;a=commit;h=4f649b43f4a96e032e15611654eff6e8bb39c64a) - thanks, @stuarthendren and @nacx!

**Pull request closed without merging**